pkg/credentialprovider/azure/azure_credentials.go

```diff
@@ -182,26 +183,26 @@ func (a *acrProvider) Enabled() bool {
 }
 
 func (a *acrProvider) Provide(image string) credentialprovider.DockerConfig {
+	klog.V(2).Infof("try to provide secret for image %s", image)
 	cfg := credentialprovider.DockerConfig{}
-	ctx, cancel := getContextWithCancel()
-	defer cancel()
 
 	if a.config.UseManagedIdentityExtension {
-		klog.V(4).Infof("listing registries")
-		result, err := a.registryClient.List(ctx)
+		// Parse login server from parameter `image`. An example of parameter `image`: foo.azurecr.io/bar/imageName:version
+		regEx, err := regexp.Compile(`.*\.azurecr\.io|.*\.azurecr\.cn|.*\.azurecr\.de|.*\.azurecr\.us`)
 		if err != nil {
-			klog.Errorf("Failed to list registries: %v", err)
-			return cfg
+			klog.Error("Unexpected error: failed to construct regular expression: %v", err)
 		}
-
-		for ix := range result {
-			loginServer := getLoginServer(result[ix])
-			klog.V(2).Infof("loginServer: %s", loginServer)
+		match := regEx.FindAllString(image, -1)
+		if len(match) != 1 {
+			klog.V(4).Infof("Found the image is not in acr, will not generate the token using MSI")
```

Comment:
klog.V(4).Infof("image(%s) is not from ACR, skip MSI authentication", image)

Reply:
Updated.